BEIMA.Backend.Test/MongoConnectorTest.cs

```diff
@@ -0,0 +1,227 @@
+﻿using NUnit.Framework;
+using BEIMA.Backend.MongoService;
+using System.IO;
+using Newtonsoft.Json;
+using System.Collections.Generic;
+using System;
+using MongoDB.Bson;
+
+namespace BEIMA.Backend.Test
+{
+    [TestFixture]
+    public class MongoConnectorTest
+    {
+        //private MongoConnector? mongo;
+        class LocalSettings
+        {
+            public bool IsEncrypted { get; set; }
+            public Dictionary<string, string> Values { get; set; }
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            //For local testing, the try block will be used to instantiate local variables.
+            //For cloud testing, this will throw an exception, and it will use the env variables
+            //defined in beima.yml.
+            try
+            {
+                string[] paths = { "..", "..", "..", "..", "BEIMA.Backend", "local.settings.json" };
+                string combinedPath = Path.Combine(paths);
+                string fullPath = Path.GetFullPath(combinedPath);
+                var settings = JsonConvert.DeserializeObject<LocalSettings>(
+                    File.ReadAllText(fullPath));
+
+                foreach (var setting in settings.Values)
+                {
+                    Environment.SetEnvironmentVariable(setting.Key, setting.Value);
+                }
+            }
+            catch (Exception ex)
+            {
+                Console.WriteLine(ex.ToString());
+            }
+            
+        }
+
+        [OneTimeTearDown]
+        public void OneTimeTearDown()
+        {
+        }
+
+        [SetUp]
+        public void SetUp()
+        {
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+        }
+
+        [Test]
+        public void InitialState_Action_ExpectedResult()
+        {
+            Assert.IsTrue(true);
+        }
+
+        [Test]
+        public void ConnectorNotCreated_CallConstructorFiveTimes_FiveInstancesAreEqual()
+        {
+            //Tests to make sure singleton is working
+            var connector1 = MongoConnector.Instance;
+            var connector2 = MongoConnector.Instance;
+            var connector3 = MongoConnector.Instance;
+            var connector4 = MongoConnector.Instance;
+            var connector5 = MongoConnector.Instance;
+            Assert.IsTrue(connector1 == connector2);
+            Assert.IsTrue(connector1 == connector3);
+            Assert.IsTrue(connector1 == connector4);
+            Assert.IsTrue(connector1 == connector5);
```

Comment:
```suggestion
            Assert.That(connector1, Is.EqualTo(connector2));
            Assert.That(connector1, Is.EqualTo(connector3));
            Assert.That(connector1, Is.EqualTo(connector4));
            Assert.That(connector1, Is.EqualTo(connector5));
```
Not a big issue, but usually it's better to use this format because it will log a much less vague error message if it fails.

Reply:
Can do this for the other ones as well. Basically, instead of saying "the assertion was false", it will say something like "connector1 was not equal to connector2"

Reply:
If we can get a more detailed log then I think that is a good change to make

Reply:
I like how this reads better too.

Reply:
Yeah, usually Assert.That(object, Is....) is my go to format for this.

Reply:
Fixed [6f34fb6](https://github.com/cs481-ekh/s22-beima/pull/67/commits/6f34fb69b240040dce07b88ac263f3df03d88033)